test/jdk/javax/imageio/plugins/jpeg/WriteJPEGThumbnailTest.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/**
+ * @test
+ * @bug     8351110
+ * @summary Test verifies that when a JFIF thumbnail may exceed 65535 bytes
+ *          we still write a valid JPEG file by clipping the thumbnail.
+ * @run     main WriteJPEGThumbnailTest
+ */
+
+import javax.imageio.IIOImage;
+import javax.imageio.ImageIO;
+import javax.imageio.ImageReader;
+import javax.imageio.ImageWriter;
+import javax.imageio.stream.ImageInputStream;
+import javax.imageio.stream.ImageOutputStream;
+import java.awt.geom.AffineTransform;
+import java.awt.image.BufferedImage;
+import java.awt.Color;
+import java.awt.Graphics2D;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.Iterator;
+import java.util.List;
+
+public class WriteJPEGThumbnailTest {
+
+    private static void assertEquals(int expected, int observed) {
+        if (expected != observed) {
+            throw new Error("expected " + expected + ", but observed " + observed);
+        }
+    }
+
+    public static void main(String[] args) throws Exception {
+        // this always passed. 21800 * 3 = 65400, which fits in a 65535-byte segment.
+        boolean b1 = new WriteJPEGThumbnailTest(100, 218).run();
+
+        // this failed prior to resolving 8351110. 21900 * 3 = 65700, which is too large
+        // for a JPEG segment. Now we clip the thumbnail to make it fit. (Previously
+        // we wrote a corrupt JPEG file.)
+        boolean b2 = new WriteJPEGThumbnailTest(100, 219).run();
+        if (!(b1 && b2))
+            System.err.println("Test failed");
```

Comment:
```suggestion
        if (!(b1 && b2)) {
            System.err.println("Test failed");
            throw new Error("Test failed");
        }
```
Always use braces, even for one-line statement.

You must throw an exception, otherwise the test will never fail from jtreg point of view.

```text
jdk> ./build/windows-x86_64-server-release/jdk/bin/java -jar ../jtreg/lib/jtreg.jar \
    -w:./build/jtwork -nr -v \
    test/jdk/javax/imageio/plugins/jpeg/WriteJPEGThumbnailTest.java
runner starting test: javax/imageio/plugins/jpeg/WriteJPEGThumbnailTest.java
runner finished test: javax/imageio/plugins/jpeg/WriteJPEGThumbnailTest.java
Passed. Execution successful
Test results: passed: 1
```
And I don't have your fix.

Reply:
The WriteJPEGThumbnailTest.jtr says, the test failed:

```text
----------System.out:(3/74)----------
Testing thumbnail 100x218...
	Test passed
Testing thumbnail 100x219...
----------System.err:(15/1153)----------
javax.imageio.IIOException: Unsupported JPEG process: SOF type 0xc8
        at ...
        at WriteJPEGThumbnailTest.run(WriteJPEGThumbnailTest.java:92)
        at WriteJPEGThumbnailTest.main(WriteJPEGThumbnailTest.java:64)
        at ...
Test failed
```

Reply:
Thank you for pointing out the `new Error` oversight.

> The WriteJPEGThumbnailTest.jtr file says, the test failed

That output is exactly what I would expect to see before this PR. Is it possible you ran the test without invoking `make`?

When I run this test my jtr file includes:

```
----------messages:(8/301)----------
command: main WriteJPEGThumbnailTest
reason: User specified action: run main WriteJPEGThumbnailTest 
started: Sat Apr 05 01:24:48 EDT 2025
Mode: othervm
Additional options from @modules: --add-modules java.desktop
Process id: 63942
finished: Sat Apr 05 01:24:49 EDT 2025
elapsed time (seconds): 1.248
----------configuration:(3/40)----------
Boot Layer
  add modules: java.desktop

----------System.out:(4/84)----------
Testing thumbnail 100x218...
	Test passed
Testing thumbnail 100x219...
	Test passed
----------System.err:(1/15)----------
STATUS:Passed.
----------rerun:(24/1876)*----------
```

(I tested both before and after the revisions of the last 24 hrs.)

Reply:
Ah, OK. So is there any outstanding question in this thread?

Reply:
> Ah, OK. So is there any outstanding question in this thread?

No, it's resolved now. You throw an `Error` to make the test fail, and jtreg now reports a failed test without the fix.